ci-operator/step-registry/cucushift/ota/preupgrade/cucushift-ota-preupgrade-commands.sh

```diff
@@ -20,6 +20,84 @@ function extract_oc(){
     oc version --client
 }
 
+function get_tp_operator(){
+    local short_version=$1
+    case ${short_version} in
+    "4.14")
+    tp_operator=("cluster-api" "platform-operators-aggregated")
+    ;;
+    "4.15")
+    tp_operator=("cluster-api" "platform-operators-aggregated" "olm")
+    ;;
+    "4.16")
+    tp_operator=("cluster-api" "platform-operators-aggregated" "olm")
+    ;;
+    *)
+    tp_operator=()
+    ;;
+    esac
+    echo ${tp_operator[*]}
+}
+
+function check_tp_operator_notfound(){
+    local try=0 max_retries=2
+    declare -A tp_resourece=(
+        ["cluster-api"]="openshift-cluster-api"
+        ["platform-operators-aggregated"]="openshift-platform-operators"
+        ["olm"]="openshift-cluster-olm-operator"
+    )
+    tmp_log=$(mktemp)
+    while (( try < max_retries )); do
+        oc get co $1 2>&1 | tee "${tmp_log}"
+        if grep -q 'NotFound' "${tmp_log}"; then
+            oc get ns ${tp_resourece[$1]} 2>&1 | tee "${tmp_log}"
+            if grep -q 'NotFound' "${tmp_log}"; then
+                (( try += 1 ))
+                sleep 60
+                break
+            else
+                echo "Unexpected ns found for $1!"
+                return 1
+            fi
+        else
+            echo "Unexpected operator found $1!"
+            return 1
+        fi
+    done
+    if (( ${try} >= ${max_retries} )); then
+        echo "Not found the tp operator $1"
+        return 0
+    fi
+}
+
+function check_manifest_annotations(){
+    local manifest="/tmp/manifest"
```

Comment:
Suggest to use an uniq folder name, e.g: `mktemp` to avoid folder name conflicts.

Reply:
updated